components/data_comps/datm/cime_config/buildlib

```diff
@@ -1,34 +1,40 @@
-#! /usr/bin/env perl
-use strict;
-
-if ($#ARGV == -1) {
-    die " ERROR datm.buildexe: must specify a caseroot input argument";
-}
-my ($CASEROOT) = @ARGV;
-chdir "${CASEROOT}";
-
-my $model = "atm";
-my $comp  = "datm";
-
-my $CASEROOT  = `./xmlquery CASEROOT  -value`;
-my $CASETOOLS = `./xmlquery CASETOOLS -value`;
-my $CIMEROOT  = `./xmlquery CIMEROOT  -value`;
-my $MACH      = `./xmlquery MACH      -value`;
-my $OBJROOT   = `./xmlquery OBJROOT   -value`;
-my $LIBROOT   = `./xmlquery LIBROOT   -value`;
-my $GMAKE_J   = `./xmlquery GMAKE_J   -value`;
-my $GMAKE     = `./xmlquery GMAKE     -value`;
-
-chdir "${OBJROOT}/${model}/obj";
-
-open(file,">Filepath") or die "Could not open file Filepath to write";
-print file "${CASEROOT}/SourceMods/src.${comp}\n";
-print file "${CIMEROOT}/components/data_comps/${comp}\n";
-close(file);
-
-my $sysmod = "$GMAKE complib -j ${GMAKE_J} MODEL=${model} COMPLIB=${LIBROOT}/lib${model}.a -f ${CASETOOLS}/Makefile MACFILE=${CASEROOT}/Macros.${MACH}";
-system($sysmod) == 0 or die "ERROR: $sysmod failed: $?\n";
-
-exit(0);
+#!/usr/bin/env python2
+
+"""
+build data model library
+"""
+
+import sys, os
+
+_CIMEROOT = os.environ.get("CIMEROOT")
+if _CIMEROOT is None:
+    raise SystemExit("ERROR: must set CIMEROOT environment variable")
+sys.path.append(os.path.join(os.environ["CIMEROOT"],"scripts","Tools"))
+
+from standard_script_setup import *
+from CIME.buildlib         import build_data_lib, parse_command_line
+
+logger = logging.getLogger(__name__)
+
+###############################################################################
+def _main_func():
+###############################################################################
+
+    if "--test" in sys.argv:
+        test_results = doctest.testmod(verbose=True)
+        sys.exit(1 if test_results.failed > 0 else 0)
+
+    parser = argparse.ArgumentParser()
+
+    CIME.utils.setup_standard_logging_options(parser)
+
+    caseroot, bldroot, libroot = parse_command_line(sys.argv, parser)
+
+    build_data_lib('atm', 'datm', caseroot, bldroot, libroot)
```

Comment:
Yes - you are right - this is redundant. I'll fix this and push my changes.

On Mon, Jun 13, 2016 at 2:23 PM, James Foucar <notifications@github.com>
wrote:

> In components/data_comps/datm/cime_config/buildlib
> <https://github.com/ESMCI/cime/pull/209#discussion_r66860248>:
>
> > +
> > +###############################################################################
> > +def _main_func():
> > +###############################################################################
> > +
> > +    if "--test" in sys.argv:
> > +        test_results = doctest.testmod(verbose=True)
> > +        sys.exit(1 if test_results.failed > 0 else 0)
> > +
> > +    parser = argparse.ArgumentParser()
> > +
> > +    CIME.utils.setup_standard_logging_options(parser)
> > +
> > +    caseroot, bldroot, libroot = parse_command_line(sys.argv, parser)
> > +
> > +    build_data_lib('atm', 'datm', caseroot, bldroot, libroot)
>
> Is the second argument redundant? The function is "build_*data*_lib" , so
> should it already know to do 'datm' if 'atm' is the first argument?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ESMCI/cime/pull/209/files/9fb22035762d87ebb3f02ab54aec7b437f13c376#r66860248>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHlxExiHZAlyMLCdMwJkpWaEsZC8RS-Cks5qLbxHgaJpZM4IyM96>
> .
>